src/java/org/apache/cassandra/hints/HintsDescriptor.java

```diff
@@ -81,18 +81,26 @@ final class HintsDescriptor
     final UUID hostId;
     final int version;
     final long timestamp;
+    final String hintsFileName;
+    final String crc32FileName;
 
     final ImmutableMap<String, Object> parameters;
     final ParameterizedClass compressionConfig;
 
     private final Cipher cipher;
     private final ICompressor compressor;
 
+    // size of underlying hint file (without descriptor header and crc file sizes)
+    // the value is populated in Session's close method in HintsWriter
+    volatile long size = 0;
```

Comment:
`size` -> `fileSize` slightly more descriptive. May also mark it as `transient`, to borrow the Accord convention (for the reader, not for the JVM).

Reply:
why not